introspection-engine/introspection-engine-tests/tests/tables/mod.rs

```diff
@@ -0,0 +1,697 @@
+        }
+    "#};
+
+    assert_eq_datamodels!(dm, &api.introspect().await?);
+
+    Ok(())
+}
+
+#[test_each_connector(tags("mssql_2017", "mssql_2019"))]
+async fn a_table_with_two_partial_indices_in_conjunct_should_be_ignored(api: &TestApi) -> crate::TestResult {
+    api.barrel()
+        .execute(move |migration| {
+            migration.create_table("pages", move |t| {
+                t.add_column("id", types::primary());
+                t.add_column("static", types::integer().nullable(false));
+                t.add_column("other", types::integer().nullable(false));
+
+                t.add_partial_index(
+                    "partial",
+                    types::index(vec!["static", "other"]).unique(true),
+                    "static IS NOT NULL AND other > 100",
+                );
+            });
+        })
+        .await?;
+
+    let dm = indoc! {r#"
+        model pages {
+            id     Int     @id @default(autoincrement())
+            static Int
+            other  Int
+        }
+    "#};
+
+    assert_eq_datamodels!(dm, &api.introspect().await?);
+
+    Ok(())
+}
+
+#[test_each_connector(tags("postgres"))]
+async fn introspecting_a_table_with_json_type_must_work(api: &TestApi) -> crate::TestResult {
+    api.barrel()
+        .execute(|migration| {
+            migration.create_table("Blog", |t| {
+                t.add_column("id", types::primary());
+                t.add_column("json", types::json());
+            });
+        })
+        .await?;
+
+    let dm = indoc! {r#"
+        datasource postgres {
+            provider = "postgres"
+            url = "postgresql://asdlj"
+        }
+
+        model Blog {
+            id      Int @id @default(autoincrement())
+            json    Json
+        }
+    "#};
+
+    let expected = format!(
+        r#"
+        datasource postgres {{
+            provider = "postgres"
+            url = "postgresql://asdlj"
+        }}
+
+        {}
+    "#,
+        api.introspect().await?
+    );
+
+    assert_eq_datamodels!(dm, &expected);
+
+    Ok(())
+}
+
+#[test_each_connector]
+async fn a_table_without_uniques_should_be_commented_out(api: &TestApi) -> crate::TestResult {
+    api.barrel()
+        .execute(|migration| {
+            migration.create_table("User", |t| {
+                t.add_column("id", types::primary());
+            });
+
+            migration.create_table("Post", |t| {
+                t.add_column("id", types::integer());
+                t.add_column("user_id", types::integer());
+                t.add_foreign_key(&["user_id"], "User", &["id"]);
+            });
+        })
+        .await?;
+
+    let dm = "// The underlying table does not contain a valid unique identifier and can therefore currently not be handled.\n// model Post {\n  // id      Int\n  // user_id Int\n  // User    User @relation(fields: [user_id], references: [id])\n// }\n\nmodel User {\n  id      Int    @id @default(autoincrement())\n  // Post Post[]\n}\n";
```

Comment:
indoc

Reply:
This test was actually a duplicate from `commenting_out/mod.rs`. Removing this.